libs/libdaemon/include/daemon/BaseDaemon.h

```diff
@@ -142,6 +144,9 @@ class BaseDaemon : public Poco::Util::ServerApplication
 	/// thread safe
 	virtual void onInterruptSignals(int signal_id);
 
+	template <class Daemon>
+	static boost::optional<Daemon &> tryGetInstance();
```

Comment:
std::experimental::optional doesn't support references.
http://stackoverflow.com/questions/26858034/stdoptional-specialization-for-reference-types

this class already using boost::optional in getLayer.